### ISSUE TITLE
Overloaded translation

### ISSUE THREAD

**Issue by forxer** [Laravel-Lang/publisher#329]
### Environment

- Laravel-Lang Publisher Version: 14.6.4
- Laravel Version: 10.9.0
- PHP Version: 8.2

### Issue description

Hello,

I'm experiencing strange behavior with my [forxer/generic-term-translations-for-laravel](https://github.com/forxer/generic-term-translations-for-laravel) plugin for Laravel Lang

I have a file /source/misc.php in which there is an array key named "required"' [Link](https://github.com/forxer/generic-term-translations-for-laravel/blob/010719311012de59d78e8b4236db4df8388d8233/source/misc.php#L27)

When I update the language files, I find /lang/{language}/misc.php with the good translation.

So far, so good

But when I call the translation `{{ trans('misc.required') }}` this is replaced by that of the validation.php file.

I was expecting to get what's in the plugin's misc.php file but I got what's in the validation.php file instead.

### Steps to reproduce

Install [forxer/generic-term-translations-for-laravel](https://github.com/forxer/generic-term-translations-for-laravel)

Display in a view {{ trans('misc.required') }}

Expected string:

- "required" in en 
- "requis" in fr

Result string:

- "The :attribute field is required." in en 
- "Le champ :attribute est obligatoire." in fr

**Comment by andrey-helldar** [comment]
My steps:

1.
    ```bash
    laravel new test
    cd test
    composer require forxer/generic-term-translations-for-laravel --dev
    php artisan lang:add en fr
    ```

2. Replace `locale` option with `en` value in the `config/app.php` file
3. I see the structure of the `lang/fr/misc.php` file according to the [original](https://github.com/forxer/generic-term-translations-for-laravel/blob/main/source/misc.php). All keys are translated into French.
    
![image](https://user-images.githubusercontent.com/10347617/236240864-8f5b2c98-4347-4b4a-9d5c-5f31facea8ae.png)

4. I replaced the contents of the `resources/views/welcome.blade.php` file with:
    ```
    {{ trans('misc.required') }}
    ```

5. Call the `php artisan serve` console command and open http://127.0.0.1:8000
6. I see `requis`:
    
![image](https://user-images.githubusercontent.com/10347617/236241540-269a7c0e-23e5-428a-90da-f8b9f6431b32.png)

**Comment by andrey-helldar** [comment]
The following test also showed no problems:

1. `php artisan lang:update`
2. Call the `php artisan serve` console command and open http://127.0.0.1:8000/

![image](https://user-images.githubusercontent.com/10347617/236242994-cc3eaac6-602f-49eb-8472-cb842d798c5a.png)

Next, I publish internal files:

```bash
php artisan lang:publish
```

![image](https://user-images.githubusercontent.com/10347617/236243209-3d8c9c87-1449-4214-88eb-2312792331fc.png)

And no problem either.

![image](https://user-images.githubusercontent.com/10347617/236243251-afc3332b-fd42-482a-87e2-2feb87b8a174.png)

Next, I run several commands and already here the problem was found:

```bash
composer require laravel-lang/common
php artisan lang:update
php artisan serve
```

![image](https://user-images.githubusercontent.com/10347617/236244237-556ecc70-8c6c-4114-bfd1-f9fb0bade3bc.png)

![image](https://user-images.githubusercontent.com/10347617/236244278-f12a111c-3056-4527-adcd-0d0ca75d7a24.png)

The problem is that translation files contain common values regardless of the source file. Thus, if the `misc` and `variations` files contain the same keys (in our example, it is "required"), then in the end the translation will be taken from the "variations" file, because it comes after "misc" alphabetically, and the "array_merge" function overwrites the previous values.

This was done specifically to reduce the amount of translation, because. identical keys generally expect the same values. But sometimes there are such collisions.

**Comment by andrey-helldar** [comment]
If you take a look at the [php.json](https://github.com/forxer/generic-term-translations-for-laravel/blob/main/locales/fr/php.json) file, you will notice that it contains all the keys from the php files in the source folder.

When updating localizations in an application, the publisher first downloads all available plugins containing translations, compiles a map of localizations, loads the necessary keys, and only after that processes the files, extracting the necessary translations from them.

Answering the question why then not to separate the work of plugins so that each is responsible for their own translations, I’ll say that we used to have it like this, and for each plugin we had to wait for the translation, and in this mechanism, if the string is already translated in another plugin, then it will be taken and for the one where this translation is not available. Thus, we increase the coverage of translation files.

The only working solution at the moment is to use a prefix when forming the name, or wrap them in an additional key.

For example:

```
// source/misc.php
<?php

return [
    'labels' => [
        'required' => 'required',
    ],
];
```
```blade
{{ trans('misc.labels.required') }}
```

Yes, it doesn't look very elegant, but it's guaranteed to work.
